### PR TITLE
Survey connection prelimary / Registration completion

### PR DESCRIPTION
**Pull request by untra**
Lots of progress!
Registration, :heavy_check_mark: Login, :heavy_check_mark: Logout, :heavy_check_mark: Prelimary silliness for importing surveys, :heavy_check_mark:
- Created a `BaseHandler` by which all other handlers inherit. [Tornado guides](http://tornado.readthedocs.org/en/latest/guide/security.html#user-authentication) recommended this approach, suggesting all signed-in user related methods and cookie storage living there. Good call.
- in the handlers `self.current_user` can be called, which fetches and stores the database entry for the user. This is actually stored in a cookie, and refreshed at dashboard GET requests 
- moved `services.surveys` into `handlers.survey_handler`
- cleaned up code :bread: 
- mixed ideas on how to approach the survey situation in the backend. [This article on how reddit approached their dope database is required reading for backend team. ](http://highscalability.com/blog/2010/5/17/7-lessons-learned-while-building-reddit-to-270-million-page.html) . I want to apologize to the frontend team; you guys are waiting on getting survey data piped in, and the backend approach is still up in the air (for me at least. But that's why I'm apoligizing :anguished: ) . Still, prelimary work was set up. This is something to discuss this Thursday.

### CODE REVIEW

File: src/config/config.py [hunk 1]
```diff
@@ -14,8 +14,9 @@
     'autoreload': True,
     'template_path':'templates/',
     'static_path':'static/',
+    'login_url': '/login'
```
Comment (nicot) [inline src/config/config.py:17]: Why is this route in a different place than the other routes?

File: src/handlers/dashboard_handler.py [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+class DashboardHandler(BaseHandler):
 
-class DashboardHandler(tornado.web.RequestHandler):
+    # @tornado.web.authenticated
     def get(self):
-        self.render('dashboard.html')
\ No newline at end of file
+        self.refresh_current_user_cookie()
+        page_json = self.current_user
+        self.render('dashboard.html', page_json=page_json)
```
Comment (nicot) [inline src/handlers/dashboard_handler.py:10]: Shouldn't this be something that react handles?

File: src/config/config.py (hunk 1 above)
Comment (untra) [inline src/config/config.py:17]: That is not a route! See [this](http://guillaumevincent.com/2013/02/12/Basic-authentication-on-Tornado-with-a-decorator.html). That is a configuration option that tells Tornado sitewide login is at `/login`

**Comment by untra** [comment]
I thought we merged this. apparently not.

**Pull request merged**